sunpy/physics/differential_rotation.py

```diff
@@ -142,3 +146,134 @@ def solar_rotate_coordinate(coordinate, new_observer_time, new_observer_location
+    rotated_coord.cache.clear()
+
+    # Go back to pixel co-ordinates
+    x2, y2 = smap.world_to_pixel(rotated_coord)
+    # Re-stack the data to make it correct output form
+    xy2 = np.dstack([x2.T.value.flat, y2.T.value.flat])[0]
+    # Returned a masked array with the non-finite entries masked.
+    xy2 = np.ma.array(xy2, mask=np.isnan(xy2))
+    return xy2
+
+
+@u.quantity_input(dt=u.s)
+def diffrot_map(smap, dt, pad=False):
+    """
+    Function to apply solar differential rotation to a sunpy map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map`
+        Original map that we want to transform.
+    dt : `~astropy.units.Quantity`
+        Desired interval between the input map and returned map.
+    pad : `bool`
+        Whether to create a padded map for submaps to don't loose data
+
+    Returns
+    -------
+   `~sunpy.map`
+        A map with the result of applying solar differential rotation to the
+        input map.
+    """
+    new_time = smap.date + timedelta(seconds=dt.to(u.s).value)
+
+    # Check for masked maps
+    if smap.mask is not None:
+        smap_data = np.ma.array(smap.data, mask=smap.mask)
+    else:
+        smap_data = smap.data
+
+    submap = False
+    # Check whether the input is a submap
+    if ((2 * smap.rsun_obs > smap.top_right_coord.Tx - smap.bottom_left_coord.Tx) or
+        (2 * smap.rsun_obs > smap.top_right_coord.Ty - smap.bottom_left_coord.Ty)):
+        submap = True
+        if pad:
+            # Calculating the largest distance between the corners and their rotation values
+            bottomleft_rotated = solar_rotate_coordinate(smap.bottom_left_coord, new_time)
+            bottom_distance = bottomleft_rotated.separation(smap.bottom_left_coord) / max(smap.scale)
+            topright_rotated = solar_rotate_coordinate(smap.top_right_coord, new_time)
+            top_distance = topright_rotated.separation(smap.top_right_coord) / max(smap.scale)
+            padding = max(bottom_distance.to(u.pix), top_distance.to(u.pix))
+            padding = np.int(np.ceil(padding.value))
+            # Create a new `smap` with the padding around it
+            smap_data = np.pad(smap.data, (padding, padding), 'constant', constant_values=(0, 0))
+            smap_meta = deepcopy(smap.meta)
+            smap_meta['naxis2'], smap_meta['naxis1'] = smap_data.shape
+            smap_meta['crpix1'] += padding
+            smap_meta['crpix2'] += padding
+            smap = sunpy.map.Map(smap_data, smap_meta)
+
+    # Apply solar differential rotation as a scikit-image warp
+    out = transform.warp(to_norm(smap_data), inverse_map=_warp_sun_coordinates,
+                         map_args={"smap": smap, "dt": dt})
+
+    # Recover the original intensity range.
+    out = un_norm(out, smap.data)
+
+    # Update the meta information with the new date and time, and reference pixel.
+    out_meta = deepcopy(smap.meta)
+    date_keys = ('date-obs', 'date_obs')
+    for k in date_keys:
```

Comment:
You could just delete both these keys and add a date-obs which I think is the standard one.